pkg/performanceprofile/controller/utils.go

```diff
@@ -0,0 +1,51 @@
+package controller
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	olmoperators "github.com/operator-framework/api/pkg/operators/install"
+	olmv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+func ListPerformanceOperatorCSVs(k8sclient client.Client, options []client.ListOption, paginationLimit uint64) ([]olmv1alpha1.ClusterServiceVersion, error) {
+	// Register OLM types to the client
+	olmoperators.Install(k8sclient.Scheme())
+
+	var performanceOperatorCSVs []olmv1alpha1.ClusterServiceVersion
+
+	csvs := &olmv1alpha1.ClusterServiceVersionList{}
+	initialOpts := append(options, client.Limit(paginationLimit))
+	opts := initialOpts
+
+	for {
+		if err := k8sclient.List(context.TODO(), csvs, opts...); err != nil {
+			if !errors.IsNotFound(err) {
+				return performanceOperatorCSVs, err
+			}
+		}
+		continueToken := csvs.GetContinue()
+
+		for i := range csvs.Items {
+			csv := &csvs.Items[i]
+			deploymentSpecs := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
+
+			for _, deployment := range deploymentSpecs {
+				if deployment.Name == "performance-operator" {
+					performanceOperatorCSVs = append(performanceOperatorCSVs, *csv)
+					break
+				}
+			}
+
+		}
+		if continueToken == "" {
+			// empty token means there is no more elements
+			break
+		}
+		// add new continuation token and keep looking
+		opts = append(initialOpts, client.Continue(continueToken))
```

Comment:
append or replace? can we end up with more than 1 token in the options?

Reply:
AFAIU we can't because I am using `initialOpts` , which do not have the token, as "base" to append the token
A different thing is that maybe using replace could be better or at least more easy to read.

Reply:
About the mutability scenario I was thinking on use `copy` to avoid it.
	opts := []client.ListOption{}
	copy(opts, options)
	opts = append(options, client.Limit(paginationLimit))
```

Also I think that even using just `opts` and appending new `Continue` tokens at the end will work, as the `List` method will use `ApplyOptions` to merge all the options into just one applying the last of them over the previous so the values of the latest elements would prevail. What I am not sure of is if we can rely on this or it is just an implementation detail